### PR TITLE
Guava

### PR DESCRIPTION
**Pull request by thespags**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -67,18 +67,24 @@
     </licenses>
 
     <properties>
-        <maven-compiler-plugin.version>3.2</maven-compiler-plugin.version>
+        <!-- Java version -->
```
Comment (thespags) [inline pom.xml]: Seems like we should have a spals parent pom that includes all of our common libraries we like to use. Thoughts?

File: src/main/java/net/spals/midas/serializer/Strings.java [hunk 2]
```diff
@@ -30,56 +30,69 @@
...
-    @VisibleForTesting
-    static final Map<String, WeakReference<byte[]>> STRING_TO_BYTES = new WeakHashMap<>();
-    @VisibleForTesting
-    static final Map<byte[], WeakReference<String>> BYTES_TO_STRING = new WeakHashMap<>();
+    private static final Strings INSTANCE = new Strings(Ticker.systemTicker());
+    private final Cache<String, byte[]> stringToBytes;
+    private final Cache<byte[], String> bytesToString;
 
-    private Strings() {
+    Strings(final Ticker ticker) {
```
Comment (thespags) [inline src/main/java/net/spals/midas/serializer/Strings.java]: Ticker allows me to unit test the cache.

File: src/main/java/net/spals/midas/serializer/Strings.java [hunk 3]
```diff
@@ -30,56 +30,69 @@
...
 /**
- * Attempts some basic interning for encoding, decoding strings as a bi map.
+ * Attempts some basic interning for encoding, decoding strings as a two {@link Cache}'s that expire after a few seconds.
+ * The performance is probably irrelevant for the use case but it was a thought expirament to use guava's {@link Cache}.
  *
  * @author spags
  */
 final class Strings {
 
     static final String NULL = "<null>";
```
Comment (thespags) [inline src/main/java/net/spals/midas/serializer/Strings.java]: I think null should be wrapped in a function call. What do you think?

File: pom.xml (hunk 1 above)
Comment (timkral) [inline pom.xml]: Yes.

File: src/main/java/net/spals/midas/GoldFile.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015, James T Spagnola & Timothy P Kral
+ * Copyright (c) 2016, James T Spagnola & Timothy P Kral
```
Comment (timkral) [inline src/main/java/net/spals/midas/GoldFile.java]: Just as a side note, I think there's a header plugin that helps with this sort of thing.

File: src/main/java/net/spals/midas/serializer/Strings.java (hunk 3 above)
Comment (timkral) [inline src/main/java/net/spals/midas/serializer/Strings.java]: Not sure I follow.

**Comment by timkral** [comment]
@thespags Looks like the CI is missing the GPG secret key for signing.

**Pull request merged**